### ISSUE TITLE
Update code with new Art Mode API

### ISSUE THREAD

**Issue by CGDaveMac** [tavicu/homebridge-samsung-tizen#675]
With firmware 1622, the art WebSocket api is back! Looks like some calls may need to be renamed. And you can check ArtMode status!

https://www.reddit.com/r/TheFrame/comments/1c2lgbs/art_mode_api_is_back/

_Originally posted by @CGDaveMac in https://github.com/tavicu/homebridge-samsung-tizen/issues/519#issuecomment-2053709576_

**Comment by fjwillemsen** [comment]
Is anyone working on implementing it? I'm not (yet) familiar with the homebridge-samsung-tizen codebase, not sure how and where to change things to implement this. Would be very grateful to anyone that does or can give pointers so I can do it!

**Comment by tavicu** [comment]
Right now I'm working on rewriting the plugin from scratch changing almost everything, but the most important change for the user is how the state of TV is detected.

I found a way that at the moment I could say it's 99% reliable :)

It was very hard advancing with the development since we had a new member in the family and my time was very limited.

Once I have all the base functionality I will order a Frame TV and check the "new" api and what allows us to do.

I will keep this issue open and update on the status as soon as I'll have one.

Starting to develop something right now it will be a waste of time since it will not be merged unless is something critical.

**Comment by fjwillemsen** [comment]
@tavicu Fantastic work, I'm very much looking forward to it! And congrats on the young one 😄 If you want I can help you test it, I have a Frame TV. Do you have a timeline / estimated release date?

**Comment by AdmiralMichael** [comment]
Just a question, so I have a pre 2022 TV and this plugin works sometimes. It's still the best method though. 

Anyways my question is would this new API and/or rewriting it improve the reliability for the older TVs?

**Comment by tavicu** [comment]
> Anyways my question is would this new API and/or rewriting it improve the reliability for the older TVs?

Yes, it will be compatible with all TVs that are currently supported

**Comment by jamesbull** [comment]
Another Frame user (2023) here! Happy to test for ya!

**Comment by joe-tofu** [comment]
And another Frame user – 2023 model as well – happy to be a test pilot here.

**Comment by ufkel** [comment]
I can bring a 2022 Frame to the test table. Happy to help.

**Comment by Foxfire0** [comment]
If it helps I can test new code on 2024 frame.

**Comment by dredgy** [comment]
Happy to provide funding as well as testing 🙃

**Comment by Cristov9000** [comment]
I have a 2021 Frame and 2023 Frame.  Happy to test any new code!

**Comment by raphtabe** [comment]
Hello ! 
Thanks for the work !
Is there anupdate 6 months  later ?
I have a 2024 model available for testing 🙂

**Comment by raphtabe** [comment]
(found my answer here: https://github.com/tavicu/homebridge-samsung-tizen/issues/696#issue-2542313465)
work is ongoing 🙂

**Comment by tavicu** [comment]
Hi. Just launched a new beta version which brings back the art mode api support

https://github.com/tavicu/homebridge-samsung-tizen/issues/756#issuecomment-5710995640

@CGDaveMac @fjwillemsen @AdmiralMichael @fjwillemsen @joe-tofu @ufkel @Foxfire0 @dredgy @Cristov9000 @raphtabe 

I know this thread is old but any help testing will be appreciated :)

Will speak there regarding the implementation.
